### PR TITLE
Add the loss of MarginRankingLoss for the paddle api2.0

### PR DESCRIPTION
**Pull request by wawltor**
### PR types
 Others

### PR changes
APIs

### Describe
Add the layer of MarginRankingLoss, add doc for this api

### CODE REVIEW

File: python/paddle/nn/layer/loss.py [hunk 1]
```diff
@@ -608,3 +611,120 @@ def forward(self, input, label):
...
+
+    If :attr:`reduction` set to ``'none'``, just return the origin ``margin_rank_loss``.
+
+    Parameters:
+        margin (float, optional): The margin value to add, default value is 0;
+        reduction (str, optional): Indicate the reduction to apply to the loss, the candicates are ``'none'``, ``'mean'``, ``'sum'``.If :attr:`reduction` is ``'none'``, the unreduced loss is returned; If :attr:`reduction` is ``'mean'``, the reduced mean loss is returned. If :attr:`reduction` is ``'sum'``, the reduced sum loss is returned. Default is ``'mean'``.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Shape: 
+        x: N-D Tensor, the shape is [N, *], Nis batch size and `*` means any number of additional dimensions., available dtype is float32, float64.
```
Comment (yaoxuefeng6) [inline python/paddle/nn/layer/loss.py]: N is

**Review by yaoxuefeng6 — DISMISSED** [review]
LGTM

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -72,6 +71,110 @@
 ]
 
 
+def margin_ranking_loss(input,
+                        other,
+                        target,
+                        margin=0.0,
```
Comment (TCChenlong) [inline python/paddle/nn/functional/loss.py:77]: target->label

File: python/paddle/nn/layer/loss.py [hunk 3]
```diff
@@ -569,3 +572,72 @@ def forward(self, input, label):
...
+    Parameters:
+        margin (float, optional): The margin value to add, default value is 0;
+        reduction (str, optional): Indicate the reduction to apply to the loss, the candicates are ``'none'``, ``'mean'``, ``'sum'``.If :attr:`reduction` is ``'none'``, the unreduced loss is returned; If :attr:`reduction` is ``'mean'``, the reduced mean loss is returned. If :attr:`reduction` is ``'sum'``, the reduced sum loss is returned. Default is ``'mean'``.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Shape: 
+        input: N-D Tensor, the shape is [N, *], N is batch size and `*` means any number of additional dimensions., available dtype is float32, float64.
+        other: N-D Tensor, `other` have the same shape and dtype as `input`.
+        target: N-D Tensor, target have the same shape and dtype as `input`.
+        out: If :attr:`reduction` is ``'mean'`` or ``'sum'`` , the out shape is :math:`[1]`, otherwise the shape is the same as `input` .The same dtype as input tensor.
```
Comment (TCChenlong) [inline python/paddle/nn/layer/loss.py:608]: out -> output

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by yaoxuefeng6 — APPROVED** [review]
LGTM

**Pull request merged**